Firestore/Example/Tests/Core/FSTQueryListenerTests.mm

```diff
@@ -23,23 +23,27 @@
 #import "Firestore/Source/Model/FSTDocumentSet.h"
 #import "Firestore/Source/Remote/FSTRemoteEvent.h"
 #import "Firestore/Source/Util/FSTAsyncQueryListener.h"
-#import "Firestore/Source/Util/FSTDispatchQueue.h"
 
 #import "Firestore/Example/Tests/Util/FSTHelpers.h"
 
+#include "Firestore/core/src/firebase/firestore/util/executor_libdispatch.h"
+#include "absl/memory/memory.h"
+
+using firebase::firestore::util::internal::ExecutorLibdispatch;
 using firebase::firestore::model::DocumentKeySet;
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface FSTQueryListenerTests : XCTestCase
-@property(nonatomic, strong, readonly) FSTDispatchQueue *asyncQueue;
 @end
 
-@implementation FSTQueryListenerTests
+@implementation FSTQueryListenerTests {
+  std::unique_ptr<ExecutorLibdispatch> _executor;
+}
 
 - (void)setUp {
-  _asyncQueue = [FSTDispatchQueue
-      queueWith:dispatch_queue_create("FSTQueryListenerTests Queue", DISPATCH_QUEUE_SERIAL)];
+  _executor = absl::make_unique<ExecutorLibdispatch>(
```

Comment:
A thought: should we add notes to these kinds of pointer-only-to-delay-initialization usages so that we undo it once we port to C++? No action required but it took me a moment to realize why this wasn't just directly a member of the test fixture.